pymks/fmks/correlations.py

```diff
@@ -164,20 +163,45 @@ def two_point_stats(arr1, arr2, periodic_boundary=True, cutoff=None):
 
     """
     if cutoff is None:
-        cutoff = arr1.shape[0]
-    nonperiodic_padder = lambda x: np.pad(
-        x, [(cutoff, cutoff)] * arr1.ndim, mode="constant", constant_values=0
+        cutoff = np.floor((np.min(arr1.shape[1:]) - 1) / 2)
+
+    nonperiodic_padder = sequence(
+        lambda x: np.pad(
+            x,
+            [(0, 0)] + [(cutoff, cutoff)] * (arr1.ndim - 1),
+            mode="constant",
+            constant_values=0,
+        ),
+        lambda x: x.rechunk(x.shape),
```

Comment:
This loses the original chunk information that `x` had. We need to rechunk only the axes that are being padded not all the axes. The chunks need to stay the same, especially for the sample axis. I'm working on it. We also need tests for this.

Reply:
That makes sense! We should also consider 3D cases if we are rechunking fixed axes